neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -37,7 +37,7 @@ protected override UInt256 GetKeyForItem((UInt256, DateTime) item)
         private VersionPayload version;
         private bool verack = false;
         private BloomFilter bloom_filter;
-
+        private uint StateRootSentIndex;
```

Comment:
Is it really worth it? To be fair, I don't quite understand why it's so problematic to answer a `getroots` request. Yeah, there are broken nodes on the network that request more roots than they need, but we're on the same network and we're just answering these requests and nothing bad happens, the nodes we operate are stable.

Reply:
Because of one previous bug, some deployed versions will send `getroots` like flood when the stateroot verification fails. We cannot stop the sender, but can only restrict it on the receiving end.

Reply:
The problem is related to the specific implementation
If the speed of sending messages in c#neo is less than the speed of receiving processing, the messages will accumulate in the sending queue